test-suite/unit-tests/src/utest.ml

```diff
@@ -0,0 +1,75 @@
+open OUnit
+
+(* general case to build a test *)
+let mk_test nm test = nm >: test
+
+(* common cases for building tests *)
+let mk_eq_test nm descr actual expected =
+  mk_test nm (TestCase (fun _ -> assert_equal ~msg:descr expected actual))
+
+let mk_bool_test nm descr actual =
+  mk_test nm (TestCase (fun _ -> assert_bool descr actual))
+
+let cfprintf oc = Printf.(kfprintf (fun oc -> fprintf oc "\n%!") oc)
+
+(* given test result, print message, return success boolean *)
+ (* todo: wanted to pass cprintf instead of out_ch here, but couldn't get it to typecheck *)
```

Comment:
I think `out_ch` is the right thing to pass though.

Reply:
OK.  If you're done I will do a squash.